Bug: `generic_config_updater/test_bgp_prefix.py` fails due to `Failed to remove bgp prefix v4 config`

### Is it platform specific

generic

### Importance or Severity

Medium

### Description of the bug

`generic_config_updater/test_bgp_prefix.py` is failing with:
```
            bgp_config = show_bgp_running_config(duthost, cli_namespace_prefix)
>           pytest_assert(
                not re.search(PREFIXES_V4_RE.format(community, PREFIXES_V4_DUMMY), bgp_config),
                "Failed to remove bgp prefix v4 config."
            )
E           Failed: Failed to remove bgp prefix v4 config.
```

We can see this entry isn't expected to be in the `show runningconfiguration bgp` output:
```
(Pdb) PREFIXES_V4_RE.format(community, PREFIXES_V4_DUMMY)
'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_0_COMMUNITY_empty_V4 seq \\d+ permit 10.30.0.0/16'

(Pdb) re.search(PREFIXES_V4_RE.format(community, PREFIXES_V4_DUMMY), bgp_config)
<re.Match object; span=(435, 525), match='ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_0_COMM>
```


When looking at the syslog after the fact I can see that the prefixes were cleaned up after we polled the bgp running config.
```
2026 Feb  3 21:05:15.282710 ctn101 NOTICE config: 'apply-patch' executed successfully for localhost by [{'op': 'remove', 'path': '/BGP_ALLOWED_PREFIXES'}] in thread:140060083802368

2026 Feb  3 21:05:15.630292 ctn101 INFO python[3633817]: ansible-ansible.legacy.command Invoked with _raw_params=show runningconfiguration bgp

2026 Feb  3 21:05:16.735983 ctn101 DEBUG bgp#bgpcfgd: BGPAllowListMgr::__remove_prefix_lists. af='v4' pl_names='PL_ALLOW_LIST_DEPLOYMENT_ID_0_COMMUNITY_empty_V4'
```

So after I hit the pdb if I re-query the `show runningconfiguration bgp` it will now be correctly excluding those config lines:
```
(Pdb) bgp_config2 = show_bgp_running_config(duthost, cli_namespace_prefix)
Wednesday 04 February 2026  00:56:46 +0000 (3:51:28.589)       3:53:11.376 ****
(Pdb) re.search(PREFIXES_V4_RE.format(community, PREFIXES_V4_DUMMY), bgp_config2)
(Pdb)
```

I've seen other places in this file that converted an `assert` to a `wait_until`.
E.G. https://github.com/sonic-net/sonic-mgmt/pull/19521

I think this will be another place where we should use a `wait_until`.

### Steps to Reproduce

`sudo ./run_tests.sh -n ardut -u -c generic_config_updater/test_bgp_prefix.py -x -e '--pdb'`

### Actual Behavior and Expected Behavior

Test should wait for the patch to take effect and then query the bgp running config

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/generic_config_updater/test_bgp_prefix.py
+++ b/tests/generic_config_updater/test_bgp_prefix.py
@@ -228,6 +228,7 @@ def bgp_prefix_tc1_replace(duthost, community, community_table, cli_namespace_pr
     finally:
         delete_tmpfile(duthost, tmpfile)
 
+
 def check_bgp_prefix_config_remove(duthost, community, cli_namespace_prefix):
     """ Check bgp prefix config for remove op
     """
@@ -246,6 +247,7 @@ def check_bgp_prefix_config_remove(duthost, community, cli_namespace_prefix):
     except Exception as e:
         logger.error(f"Error checking bgp prefix configuration: {e}")
         return False
+
 
 def bgp_prefix_tc1_remove(duthost, community, cli_namespace_prefix, namespace=None):
     """ Test to remove prefix config

